DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/dashboard/dashboard.controller.js

```diff
@@ -5,24 +5,68 @@
         .module('app.dashboard')
         .controller('DashboardController', DashboardController);
 
-    DashboardController.$inject = ['$q', 'logger', 'i18n'];
+    DashboardController.$inject = ['$q', '$cookies', 'logger', 'i18n', 'ActivityService', 'ACTIVITY_CONSTANTS'];
     /* @ngInject */
-    function DashboardController($q, logger, i18n) {
+    function DashboardController($q, $cookies, logger, i18n, activityService, ACTIVITY_CONSTANTS) {
         var vm = this;
 
         // add localization
-        vm.i18n = i18n;
+        vm.i18n = i18n.message;
         vm.news = {
             title: 'GitHubExtension',
             description: 'Hot Towel Angular is a SPA template for Angular developers.'
         };
-        vm.messageCount = 0;
         vm.title = 'Dashboard';
+        vm.currentPage = ACTIVITY_CONSTANTS.FIRST_PAGE;
+        vm.numberOfPages = ACTIVITY_CONSTANTS.FIRST_PAGE;
+
+        // angular ui pagination directive accepts as a parameter total amount of items.
+        // total amount of pages is calculated inside of it and is for read only
+        // GitHub api only returns amount of pages. Each page has 30 events or less.
+        // here I'm calculating amount of items manually
+        vm.totalItems = vm.numberOfPages * ACTIVITY_CONSTANTS.EXTERNAL_ACTIVITY_ITEMS_PER_PAGE;
```

Comment:
You are right I can simply write 
`vm.totalItems = ACTIVITY_CONSTANTS.EXTERNAL_ACTIVITY_ITEMS_PER_PAGE` or 
just omit it and it will be binded with new value on activate